src/lib/y2storage/partitionable.rb

```diff
@@ -259,7 +261,9 @@ def ensure_partition_table
     #
     # @return [Array<Partition>}]
     def partitions_with_id(*ids)
-      partitions.reject { |p| p.type.is?(:extended) }.select { |p| p.id.is?(*ids) }
+      # Sorting is not mandatory, but keeping the output stable looks like a
+      # sane practice.
+      partitions.sort_by(&:number).reject { |p| p.type.is?(:extended) }.select { |p| p.id.is?(*ids) }
```

Comment:
I would place the call to `sort_by` at the end for two reasons: 1) be consistent with L169 and 2) sort a potentially smaller collection (after reject and select).